pallets/claims/src/lib.rs

```diff
@@ -297,11 +305,27 @@ pub mod pallet {
 				.for_each(|(i, a)| {
 					Preclaims::<T>::insert(i, a);
 				});
+			// build expiryConfig
+			ExpiryConfig::<T>::set(self.expiry.clone())
 		}
 	}
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(current_block: BlockNumberFor<T>) {
+			// check if we have an expiry time set and have we crossed the limit
+			let expiry_config = ExpiryConfig::<T>::get();
+			if let Some(expiry_config) = expiry_config {
+				if current_block > expiry_config.0 {
+					let unclaimed_amount = Total::<T>::take();
+					frame_support::log::info!("Claims : Expiry block passed, sweeping remaining amount of {:?} to destination", unclaimed_amount);
+					CurrencyOf::<T>::deposit_creating(&expiry_config.1, unclaimed_amount);
+					// clear the expiry detail
+					ExpiryConfig::<T>::take();
```

Comment:
Are you forgetting to reset the total to 0?

Reply:
In line 320, `Total::<T>::take()` will remove the total from storage, effectively setting it to zero.
Also in the unit test (line 1495) it fails with the expected error
```rust
// all further claims should fail with PotUnderflow error since the funds have been
// emptied
			assert_noop!(
				Claims::claim(Origin::none(), 42, sig::<Test>(&frank(), &42u64.encode(), &[][..])),
				Error::<Test>::PotUnderflow
			);
```